src/main/java/io/anserini/fusion/ScoredDocsFuser.java

```diff
@@ -0,0 +1,270 @@
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.StoredField;
+
+import io.anserini.search.ScoredDocs;
+
+//replace topic wtih const
+public class ScoredDocsFuser {
+  public static final String TOPIC = "TOPIC";
+
+  /**
+   * Reads a TREC run file and returns a ScoredDocs containing the data.
+   * 
+   * @param filepath Path to the TREC run file.
+   * @throws IOException If the file cannot be read.
+   * @return A ScoredDocs object containing the data from the TREC run file.
+   */
+  public static ScoredDocs readRun(Path filepath, boolean reSort) throws IOException {
+    ScoredDocs scoredDocs = new ScoredDocs();
+    try (BufferedReader br = new BufferedReader(new FileReader(filepath.toFile()))) {
+      List<Document> lucene_documents = new ArrayList<>(); // topic
+      List<String> docids = new ArrayList<>(); // docid
+      List<Float> scores = new ArrayList<>(); // score
+      List<Integer> rank = new ArrayList<>(); // rank
+
+      String line;
+      while ((line = br.readLine()) != null) {
+        String[] data = line.split("\\s+");
+  
+        // Populate the lists with the parsed topic and docid
+        Document doc = new Document();
+        doc.add(new StoredField(TOPIC, data[0]));
+        lucene_documents.add(doc);
+        docids.add(data[2]);
+
+        // Parse RANK as integer
+        int rankInt = Integer.parseInt(data[3]);
+        rank.add(rankInt);
+
+        // Parse SCORE as float
+        float scoreFloat = Float.parseFloat(data[4]);
+        scores.add(scoreFloat);
+      }
+
+      scoredDocs.lucene_documents = lucene_documents.toArray(new Document[0]);
+      scoredDocs.docids = docids.toArray(new String[0]);
+      scoredDocs.scores = ArrayUtils.toPrimitive(scores.toArray(new Float[scores.size()]), Float.NaN);
+      scoredDocs.lucene_docids = ArrayUtils.toPrimitive(rank.toArray(new Integer[0]));
+    }
+  
+    if (reSort) {
+      ScoredDocsFuser.sortScoredDocs(scoredDocs);
+    }
+
+    return scoredDocs;
+  }
+
+  /**
+   * Rescored given ScoredDocs using the specified method.
+   *
+   * @param method  Rescore method to be applied (e.g., RRF, SCALE, NORMALIZE).
+   * @param rrfK    Parameter k needed for reciprocal rank fusion.
+   * @param scale   Scaling factor needed for rescoring by scaling.
+   * @param scoredDocs ScoredDocs object to be rescored.
+   * @throws UnsupportedOperationException If an unsupported rescore method is provided.
+   */
+  public static void rescore(RescoreMethod method, int rrfK, double scale, ScoredDocs scoredDocs) {
```

Comment:
Might need to think about how this method works, because some of the variables are applicable to only some of the methods... e.g., rrfK is only applicable to RRF.